bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/codechecker/locator/ResolutionMethodTypes.java

```diff
@@ -0,0 +1,41 @@
+package org.codechecker.eclipse.plugin.codechecker.locator;
+
+/**
+ * This enum represents the available types of CodeChecker resolution methods.
+ */
+public enum ResolutionMethodTypes {
+    PATH("PATH"), PRE("PRE"), PY("PY");
+    
+    private String value;
+    
+    /**
+    *
+    * @param def utility method for setting the default value.
+    */
+    private ResolutionMethodTypes(String def) {
+        this.value = def;
+    }
+
+    /**
+     *
+     * @return Returns the value associated with the enum.
+     */
+    public String getDefaultValue() {
+        return value;
+    }
+
+    /**
+     *
+     * @param s
+     *            The query string.
+     * @return The matching ResolutionMethodTypes if exists null otherwise.
+     */
+    public static ResolutionMethodTypes getFromString(String s) {
```

Comment:
If the enum symbol name is the same as the String it represents (for example in this case), then getFromString, and even the getDefault value is redundant. Consider using an enum without a base type, and use the toString() and valueOf(String) methods. (To further clarify: this would effectively reduce the enum definition to a 3 member plain enum).

Reply:
Darn, There is an another enum which could be simplified like this. Thanks for the remark, I'll make a patch for that other.